devel/re2/Portfile

```diff
@@ -22,23 +23,34 @@ checksums           rmd160  47471514269abd3b815358fa2e8934d4489985e3 \
                     sha256  3fe6eecd1bc4a254a0eac1f63c4af2002da6ec157986307a364fe4647bae9792 \
                     size    402998
 
-use_configure       no
-
 variant universal   {}
 
 compiler.cxx_standard   2011
-if {[string match *clang* ${configure.cxx}]} {
-    configure.ldflags-append -stdlib=${configure.cxx_stdlib}
-}
-
-build.args          CXX="${configure.cxx}" \
-                    CXXFLAGS="${configure.cxxflags} [get_canonical_archflags cxx]" \
-                    LDFLAGS="${configure.ldflags} [get_canonical_archflags cxx]" \
-                    prefix=${prefix}
 
-destroot.args       {*}${build.args}
+configure.args-append \
+                    -DBUILD_SHARED_LIBS:BOOL=ON
+
+# Fix pkgconfig and shared library versioning; see:
+# https://bugs.archlinux.org/task/67739
+# https://gist.github.com/mtorromeo/1d48de16534dc8ee29cd872f94b070e5
+
+pre-configure {
+    reinplace s|@includedir@|${prefix}/include/${name}|g \
+        ${worksrcpath}/re2.pc
+    reinplace s|@libdir@|${prefix}/lib|g \
+        ${worksrcpath}/re2.pc
+    set somajor [exec /bin/sh -c "grep '^SONAME=' ${worksrcpath}/Makefile | cut -d= -f2"]
+    # append this line to CMakeLists.txt
+    reinplace "\$ a\\
+set_target_properties(re2 PROPERTIES VERSION ${somajor}.0.0 SOVERSION ${somajor})" \
+        ${worksrcpath}/CMakeLists.txt
+}
```

Comment:
Comment: this adds shared library versioning per:
* https://bugs.archlinux.org/task/67739
* https://gist.github.com/mtorromeo/1d48de16534dc8ee29cd872f94b070e5